tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -2985,21 +2997,60 @@ protected function parse_markup_fragment( $markup ) {
 			}
 		}
 
-		return $body;
+		return $dom;
 	}
 
 	/**
-	 * Assert markup is equal.
+	 * Assert markup is equal after normalizing script tags.
 	 *
 	 * @param string $expected Expected markup.
 	 * @param string $actual   Actual markup.
 	 * @param string $message  Message.
 	 */
 	protected function assertEqualMarkup( $expected, $actual, $message = '' ) {
+		$expected_dom = $this->parse_markup_fragment( $expected );
+		$actual_dom   = $this->parse_markup_fragment( $actual );
+		foreach ( array( $expected_dom, $actual_dom ) as $dom ) {
+			$xpath = new DOMXPath( $dom );
+			/** @var DOMElement $script */
+
+			// Normalize type attribute. When missing, it defaults to text/javascript.
+			foreach ( $xpath->query( '//script[ not( @type ) ]' ) as $script ) {
+				$script->setAttribute( 'type', 'text/javascript' );
+			}
+
+			// Normalize script contents to remove CDATA wrapper.
+			foreach ( $xpath->query( '//script[ contains( text(), "<![CDATA[" ) ]' ) as $script ) {
+				$script->textContent = str_replace(
+					array(
+						"/* <![CDATA[ */\n",
+						"\n/* ]]> */",
+					),
```

Comment:
this is worrisome because it conflates syntax and content. I guess it's probably low-risk in our tests